File: node.cpp
# include "node.hpp"

Node::Node(int **puzzle, int **goal_map)
{
    // this->f = total_cost;
    // this->g = cost_so_far;
    // this->h = estimated_cost;
    this->map = puzzle;
}


int Node::get_f(void) const
{
    return (this->f);
}

int Node::get_g(void) const
{
    return (this->g);
}

int Node::get_h(void) const
{
    return (this->h);
}

int **Node::get_map(void) const
{
    return (this->map);
}

File: node.py
class Node:
    def __init__(self, map = [], id = "", parent = None, position = None):
          
        self.map = map
        self.id = id
        self.parent = parent
        self.position = position
        self.cost_so_far = 0
        self.heuristic = 0
        self.total_cost = 0